test/server/server.test.ts

```diff
@@ -0,0 +1,219 @@
+import {FastifyInstance} from 'fastify';
+import {Server} from 'http';
+import {
+  Catalog,
+  Dataset,
+  IRI,
+  Organization,
+  SparqlDistribution,
+} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
+import {server} from '../../src/server/server';
+import {gql} from 'mercurius-codegen';
+import {setup, teardown} from 'jest-dev-server';
+
+let httpServer: FastifyInstance<Server>;
+const catalog = new Catalog([
+  new Dataset(
+    new IRI('http://vocab.getty.edu/aat'),
+    'Art & Architecture Thesaurus',
+    [
+      new Organization(
+        new IRI('http://www.getty.edu/research/'),
+        'Getty Research Institute',
+        'Getty'
+      ),
+    ],
+    [
+      new SparqlDistribution(
+        new IRI('http://vocab.getty.edu/aat/sparql'),
+        new IRI('http://localhost:3000/sparql'),
+        gql`CONSTRUCT { ?s ?p ?o } WHERE { ?s ?p ?o }`
+      ),
+    ]
+  ),
+  new Dataset(
+    new IRI('https://data.cultureelerfgoed.nl/term/id/cht'),
+    'Cultuurhistorische Thesaurus',
+    [
+      new Organization(
+        new IRI('https://www.cultureelerfgoed.nl'),
+        'Rijksdienst voor het Cultureel Erfgoed',
+        'RCE'
+      ),
+    ],
+    [
+      new SparqlDistribution(
+        new IRI('https://example.com/distributions/endpoint-error'),
+        new IRI('http://does-not-resolve'),
+        gql`CONSTRUCT { ?s ?p ?o } WHERE { ?s ?p ?o }`
+      ),
+    ]
+  ),
+  new Dataset(
+    new IRI('https://data.rkd.nl/rkdartists'),
+    'RKDartists',
+    [
+      new Organization(
+        new IRI('https://rkd.nl'),
+        'RKD – Nederlands Instituut voor Kunstgeschiedenis',
+        'RKD'
+      ),
+    ],
+    [
+      new SparqlDistribution(
+        new IRI('https://example.com/distributions/timeout'),
+        new IRI('https://httpbin.org/delay/3'),
+        gql`CONSTRUCT { ?s ?p ?o } WHERE { ?s ?p ?o }`
+      ),
+    ]
+  ),
+]);
+describe('Server', () => {
+  afterAll(async () => {
+    await teardown();
+  });
+  beforeAll(async () => {
+    await startDistributionSparqlEndpoint();
+    httpServer = await server(catalog);
+  });
+
+  it('responds to GraphQL sources query', async () => {
+    const body = await query(
+      gql`
+        query {
+          sources {
+            uri
+            name
+            alternateName
+            creators {
+              uri
+              name
+              alternateName
+            }
+          }
+        }
+      `
+    );
+    expect(body.data['sources']).toHaveLength(catalog.datasets.length);
```

Comment:
Out of curiosity: is `body.data.sources` allowed, rather than `body.data['sources']`?

Reply:
Actually, yes!